opentech/apply/projects/views.py

```diff
@@ -82,6 +90,41 @@ def get_contracts(self):
         return contracts
 
 
+class PaymentsMixin:
+    def get_context_data(self, **kwargs):
+        project = self.get_object()
+
+        payments = {
+            'availabe_statuses': REQUEST_STATUS_CHOICES,
+            'not_rejected': project.payment_requests.exclude(status=DECLINED),
+            'rejected': project.payment_requests.filter(status=DECLINED),
+            'totals': self.get_totals(project),
+        }
+
+        context = super().get_context_data(**kwargs)
+        context['payments'] = payments
+        return context
+
+    def get_totals(self, project):
+        def percentage(total, value):
+            return round((value / total) * 100)
+
+        unpaid_requests = project.payment_requests.filter(Q(status=SUBMITTED) | Q(status=UNDER_REVIEW))
+        awaiting_absolute = sum(unpaid_requests.values_list('value', flat=True))
```

Comment:
Alternative?

`awaiting_absolute = unpaid_requests.aggregate(total_value=Sum('value'))['total_value']`

maybe i just like over using the ORM

Reply:
Can do, it's the same thing but pushes the calculation to the database.  At the scale I believe this application runs at the difference is negligible.  Would you like me to change it?